pydantic/main.py

```diff
@@ -112,6 +112,12 @@ def __new__(mcs, name, bases, namespace):
         annotations = namespace.get('__annotations__', {})
         # annotation only fields need to come first in fields
         for ann_name, ann_type in annotations.items():
+            # Ensure that the field's name does not shadow an existing attribute
+            for base in reversed(bases):
```

Comment:
You'll need to check for non-annotation attributes too, eg. line 131 below. Best to make this a method so the logic can be reused.

Reply:
What are non-annotation attributes? Is it the ones created with the create_model() function?

Reply:
No they're attributes with a value, eg. `foo = 123`. Read the code on the line I suggested.